src/core/api_client_parallel.py

```
@@ -1,34 +1,33 @@
+from collections import defaultdict
+from typing import Dict, List, Optional, Tuple
```

Ruff (0.12.2)

`typing.Dict` is deprecated, use `dict` instead

(UP035)